src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -6358,10 +6454,236 @@ MutationCommands StorageReplicatedMergeTree::getFirstAlterMutationCommandsForPar
     return queue.getFirstAlterMutationCommandsForPart(part);
 }
 
+
 void StorageReplicatedMergeTree::startBackgroundMovesIfNeeded()
 {
     if (areBackgroundMovesNeeded())
         background_moves_executor.start();
 }
 
+
+void StorageReplicatedMergeTree::lockSharedData(const IMergeTreeDataPart & part) const
+{
+    if (!part.volume)
+        return;
+    DiskPtr disk = part.volume->getDisk();
+    if (!disk)
+        return;
+    if (disk->getType() != DB::DiskType::Type::S3)
+        return;
+
+    zkutil::ZooKeeperPtr zookeeper = tryGetZooKeeper();
+    if (!zookeeper)
+        return;
+
+    String id = part.getUniqueId();
+    boost::replace_all(id, "/", "_");
+    String norm_path = part.relative_path;
+    boost::replace_all(norm_path, "/", "_");
+
+    String zookeeper_node = zookeeper_path + "/zero_copy_s3/shared/" + part.name + "/" + id + "/" + norm_path + "/" + replica_name;
```

Comment:
Seems like we can get rid of `norm_path` part in path.

Reply:
Fixed